lib/capybara_accessible_selectors/selectors/img.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+Capybara.add_selector(:img, locator_type: [String, Symbol]) do
+  expression_filter(:src, [String, Regexp]) do |xpath, src|
+    builder(xpath).add_attribute_conditions(src: src)
+  end
+
+  describe(:expression_filters) do |src: nil, **|
+    " expected to match src \"#{src}\"" unless src.nil?
+  end
+
+  xpath do |*|
+    XPath.descendant[
+      (XPath.attr(:role) == "img") & (
+        XPath.attr(:"aria-label") | XPath.attr(:"aria-labelledby")
+      ) |
+      XPath.self(:img)[XPath.attr(:alt)]
+    ]
+  end
+
+  locator_filter do |node, locator, exact:, **|
+    next true if locator.nil?
+
+    method = exact ? :eql? : :include?
+    if node.tag_name === "img"
+      node[:"alt"]
```

Comment:
I've had a look at https://www.w3.org/TR/html-aria/ and. https://www.w3.org/TR/html-aam-1.0/#img-element-accessible-name-computation.

It looks like an element can still have the "img" role if it is an `<img>` with no `alt` attribute, but with an accessible name from an `aria-label` or `aria-labelledby`.

It also looks like <div role="img"></div> with no accessible name counts as an image.  Although `img` is name required so probably better not to support that.

Reply:
Thanks for finding that out, I can surely add that case :-) I didn't quite fully understand your last sentence. Is there something missing, like a link or an image you intended to add?

Reply:
Just pushed the first case you mentioned.

Reply:
Sorry, last sentence was missing some backticks.

You can do `<div role="img"></div>` without an accessible name and it will considered an image in the accessibility tree.  This is unlike `<img>` which will be not be considered image in the accessibility tree without an accessible name.

However it is also a rule that an image must have an accessible name, so `<div role="img"></div>` is an image, but it also an error.  So I suppose is better not to find images without an accessible name in a testing library.  But it is not very important.

Reply:
That makes sense. Thanks for sharing!